api/backend/duckdb.py

```diff
@@ -22,55 +22,56 @@
             l_comment VARCHAR NOT NULL);"""
 
 
-def ConnectDuckDB():
-    con = duckdb.connect()
-    con.install_extension("substrait")
-    con.load_extension("substrait")
-    con.install_extension("tpch")
-    con.load_extension("tpch")
-    con.execute(query=schema_lineitem)
-    logger.success("DuckDb initialized successfully")
-    return con
+class DuckDBConnection:
+    def __init__(self):
+        self.connection = None
 
+    def connect(self):
+        self.connection = duckdb.connect()
+        self.connection.install_extension("substrait")
+        self.connection.load_extension("substrait")
+        self.connection.install_extension("tpch")
+        self.connection.load_extension("tpch")
+        self.connection.execute(query=schema_lineitem)
+        logger.success("DuckDb initialized successfully")
 
-def CheckDuckDBConnection(con):
-    status = {"db_health": "unavailable"}
-    try:
-        con.execute(query="SHOW TABLES;").fetchall()
-        status["db_health"] = "up and running"
-    except Exception as e:
-        logger.info(
-            "Healthcheck failed for DuckDB"
-            "Initializing new connection object. Details: ",
-            str(e),
-        )
-    finally:
-        return status
+    def check(self):
+        status = {"db_health": "unavailable"}
+        try:
+            self.connection.execute(query="SHOW TABLES;").fetchall()
+            status["db_health"] = "up and running"
+        except Exception as e:
+            logger.info(
+                "Healthcheck failed for DuckDB"
+                "Initializing new connection object. Details: ",
```

Comment:
This comment makes it seems like we try to initialize a new connection if the health check failed.  That doesn't seem to be the case here right?